### DISCUSSION TITLE
Expected behaviour of `render` when passing a `nil`-valued parameter?

### DISCUSSION THREAD

**Discussion by JacobBeaudry** [harttle/liquidjs#418] [Q&A]
I was wondering what the expected behaviour is, if one passes a `nil` value to a parameter, in `render`? e.g. if `file123` is:
```
{% assign foo = foo | default: 'hello world' %}
This is foo in file123: {{ foo }}
```
and one then renders `file123`, passing `foo` as a `nil`-valued parameter, e.g. in parent_file1:

```
{% render "file123", foo: nil %}
```
Should it return
```
This is foo in file123: hello world
```
or 
```
This is foo in file123:
```
? 

And should anything change if, instead, we had, e.g. parent_file2 (in which the parameter `foo` is assigned to the value of the variable `foo` in parent_file2, in which it is not assigned (&, IINM, therefore defaults to `nil`)?
```
{% comment %} foo is not assigned anywhere in parent_file2 {% endcomment %}
{% render "file123", foo:foo %}
```

Hope this is clear. Thank you in advance for your time and help!

**Comment by harttle** [comment]
Scopes are inherited through `include` and `layout` calls, can be visualized as calling tree like:

```
entry.liquid
    |- include partial.liquid
    |- layout parent.liquid
        |- then include partial2.liquid
```

Each node will inherit scopes set by its parent nodes in this tree, lower parent will hide higher parent's variables with same name. Though `render` is special in this, template introduced by `render` can only access the variables passed to the tag explicitly. There should be no difference in this between `assign`ed variables and variables in initial scope, no matter whether the variable is set to `nil` or anything else.

There maybe a bug for `assign`ed variables here, feel free if you can reproduce.
